File: #Task-03/Ruby/Sub-Task3/SubTask3.rb
puts "Enter n"
n=gets.to_i
if n%2==1
  x=(n.to_f/2).floor
  c=n-x-1
  puts c
  m=1
  s1=-1
  s2=1
  for g in 0..n-1 do
    if c!=0
      print " "*c
    end
    puts "*"*m
    m+=s2*2
    c+=s1
    if c==0
      s1=1
    end
    if m>n
      s2=-1
      m=n-2
    end
  end
else
  puts "Enter Odd"
end

File: #Task-03/C/Sub Task 3/SubTask3.c
#include <stdio.h>
#include <math.h>
void main() {
    int n;
    printf("Enter Number: ");
    scanf("%d",&n);
    if(n%2!=0){
        int x=floor(n/2);
        int c=n-x-1;
        int m=1;
        int s1=-1;
        int s2=1;
        for (int g=0;g<n;g++){
            if(c!=0){
                for(int j=0;j<c;j++){
                   printf(" ");
                }
            }
            for (int j=0;j<m;j++){
                printf("*");
            }
            printf("\n");
            m+=s2*2;
            c+=s1;
            if(c==0){
                s1=1;
            }
            if (m>n){
                s2=-1;
                m=n-2;
            }
        }
    } else {
        printf("Enter Odd Number");
    }
}
